code/ai/ai.cpp

```diff
@@ -82,22 +81,23 @@ void set_wingnum(int objnum, int wingnum)
 	Ai_info[ai_index].wing = wingnum;
 }
 
-char *ai_get_goal_target_name(const char *name, int *index)
+const char *ai_get_goal_target_name(const char *name, int *index)
 {
-	Assert(name != NULL);
-	Assert(index != NULL);
-	int i;
+	Assertion(name != nullptr && index != nullptr, "Arguments cannot be null!");
 
-	for (i=0; i < Total_goal_target_names; i++)
-		if (!stricmp(name, Goal_target_names[i])) {
+	for (int i = 0; i < static_cast<int>(Goal_target_names.size()); i++) {
+		if (!stricmp(name, Goal_target_names[i].c_str())) {
 			*index = i;
-			return Goal_target_names[i];
+			return Goal_target_names[i].c_str();
```

Comment:
As far as I can tell, the returned value will be stored for an extended time which will lead to issues if the vector is resized. In that case the string data might be moved to a different pointer and the old one will become invalid.

A quick fix for this would be to allocate the `SCP_string`s dynamically then they themselves wouldn't be reallocated.

Reply:
Oh, good point.

Reply:
I used `vm_strdup` because it fit right in with the rest of the API.